src/main/java/com/conveyal/datatools/manager/controllers/api/NoteController.java

```diff
@@ -11,158 +11,151 @@
+        return objectWithNote.retrieveNotes(isProjectAdmin(userProfile, feedSource));
+    }
 
-        switch (type) {
-            case FEED_SOURCE:
-                model = Persistence.feedSources.getById(objectId);
-                break;
-            case FEED_VERSION:
-                model = Persistence.feedVersions.getById(objectId);
-                break;
-            default:
-                // this shouldn't ever happen, but Java requires that every case be covered somehow so model can't be used uninitialized
-                logMessageAndHalt(req, 400, "Unsupported type for notes");
+    public static Note createNote (Request req, Response res) throws IOException {
+        // Note: Halt will handle early return if userProfile is null.
+        Auth0UserProfile userProfile = req.attribute("user");
+        Model objectWithNote = getObjectWithNote(req);
+        // Note: Halt will handle early return if objectWithNote is null.
+        FeedSource feedSource = getFeedSourceForModel(objectWithNote);
+        // Ensure that user has create note permission.
+        checkPermissions(req, feedSource);
+        Note note = new Note();
+        note.storeUser(userProfile);
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode node = mapper.readTree(req.body());
+        note.body = node.get("body").asText();
+
+        note.userEmail = userProfile.getEmail();
+        note.date = new Date();
+        note.type = getTypeFromRequest(req);
+        note.adminOnly = node.get("adminOnly").asBoolean();
+        if (note.adminOnly && !isProjectAdmin(userProfile, feedSource)) {
+            logMessageAndHalt(
+                req,
+                HttpStatus.UNAUTHORIZED_401,
+                "User must be admin to create admin-only note."
+            );
         }
+        // Create note and update feed source or version with noteIds
+        Persistence.notes.create(note);
+        MongoCollection collection = objectWithNote instanceof FeedSource
+            ? Persistence.feedSources.getMongoCollection()
+            : Persistence.feedVersions.getMongoCollection();
+        collection.updateOne(eq(objectWithNote.id), push("noteIds", note.id));
+        String message = String.format(
+                "%s commented on %s at %s:<blockquote>%s</blockquote>",
+                note.userEmail,
+                feedSource.name,
+                note.date.toString(),
+                note.body);
+        // Send notifications to comment subscribers.
+        // TODO: feed-commented-on has been merged into feed-updated subscription type. This should be clarified
+        //  in the subject line/URL of the notification email.
+        NotifyUsersForSubscriptionJob.createNotification(
+            "feed-updated",
+            feedSource.id,
+            message
+        );
+        return note;
+    }
 
-        FeedSource s;
+    private static boolean isProjectAdmin(Auth0UserProfile userProfile, FeedSource feedSource) {
```

Comment:
Would this maybe make more sense in `userProfile` like `canViewFeed`? Could then use it lots of other places as well

Reply:
Addressed in 27187c5eb4ec40c7d0862a83f0d358cb7e355d80